File: api/app/Models/ApiToken.php
<?php
/**
 * Created by PhpStorm.
 * User: gry
 * Date: 09/07/2019
 * Time: 14:20
 */
namespace App\Models;

use Carbon\Carbon;
use Ramsey\Uuid\Uuid;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

/**
 * Class ApiToken
 * @package App\Models
 *
 * @property integer $id
 * @property string $user_id
 * @property string $token
 * @property Carbon $created_at
 * @property Carbon $expired_at
 * @property Carbon $updated_at
 * @property Carbon $deleted_at
 *
 * @property-read User $user
 */

class ApiToken extends Model {

    protected $table = 'api_token';

    protected $dates = ['expired_at'];

    use SoftDeletes;

    function __construct(array $attributes = [])
    {
        parent::__construct($attributes);

        $this->attributes['token'] = Uuid::uuid4();
    }

    function user() {
        return $this->belongsTo('App\Models\User', 'user_id', 'id');
    }

    function addTime($remember = false) {
        if ($remember) {
            $this->attributes['expired_at'] = Carbon::now()->addMonth();
        } else {
            $this->attributes['expired_at'] = Carbon::now()->addMinutes(30);
        }
    }
}

File: go-api/Model/ApiToken.go
package Model

import (
	"github.com/jinzhu/gorm"
	uuid "github.com/satori/go.uuid"
	"time"
)

type ApiToken struct {
	gorm.Model

	UserId     uuid.UUID  `gorm:"not null"`
	Token      uuid.UUID  `gorm:"unique;not null"`
	ExpiredAt  time.Time

	User       User       `gorm:"foreignKey:UserId;association_foreignKey:ID"`
}

func (apiToken ApiToken) TableName() string {
	return "api_token"
}

func (apiToken *ApiToken) BeforeCreate(scope *gorm.Scope) error {
	_ = scope.SetColumn("Token", interface{}(uuid.NewV4()))
	return nil
}

func (apiToken *ApiToken) AddTime(remember bool)  {
	now := time.Now()
	duration, _ := time.ParseDuration("30m")
	if remember {
		apiToken.ExpiredAt = now.AddDate(0, 1, 0)
	} else {
		apiToken.ExpiredAt = now.Add(duration)
	}
}